gammapy/datasets/map.py

```diff
@@ -2694,6 +2723,46 @@ def to_map_dataset(self, name=None):
             meta_table=self.meta_table,
         )
 
+    def _to_asimov_dataset(self):
+        """Create Asimov dataset from the current models.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the new dataset. Default is None.
+
+        """
+        npred = self.npred()
```

Comment:
Is `self.counts` is None, this will not work 
```
1 dataset_on_off.counts = None
----> 2 dataset_on_off.npred()

File ~/anaconda3/envs/gammapy-1.2/lib/python3.10/site-packages/gammapy/datasets/map.py:639, in MapDataset.npred(self)
    636 npred_total = self.npred_signal()
    638 if self.background:
--> 639     npred_total += self.npred_background()
    640 npred_total.data[npred_total.data < 0.0] = 0
    641 return npred_total
```

Reply:
I think empty counts should be given has an empty map so the geom is taken properly